docs/roles/integrator/exchange-integration.md

```diff
@@ -110,6 +110,390 @@ result of the ones we provided. We expect that, at this stage, most of the contr
+          args: 'eyJyZXF1ZXN0Ijp7InJlY2VpdmVyX2lkIjoibWlrZXB1cnZpcy50ZXN0bmV0IiwiYWN0aW9ucyI6W3sidHlwZSI6IlRyYW5zZmVyIiwiYW1vdW50IjoiMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMCJ9XX19',
+          gas: 30000000000000,
+          deposit: '0'
+        }
+      },
+      [length]: 1
+    ],
+    signature: 'ed25519:27bGcHP4m9SGdXwnzLtvpiFC4iXorNCS77wRcHZnnns5CDxkyfdgCs9KzvEnbSRiR1LQ2T5jNUpeC6SWBCctCyxc',
+    hash: 'Fdb4vJf6mfnaCsor6VJ63cyT415wwMKUMcQ7dE6NC21V'
+  },
+  transaction_outcome: {
+    proof: [
+      {
+        hash: '2RnN1do55cCCBWP96sKe2Wx5AEJ39C3qo1pupeuozLYu',
+        direction: 'Right'
+      },
+      [length]: 1
+    ],
+    block_hash: 'GgmsAbDsSjNK25xiV4hnMbLhWtSMzQ36GhnS7C7arN4u',
+    id: 'Fdb4vJf6mfnaCsor6VJ63cyT415wwMKUMcQ7dE6NC21V',
+    outcome: {
+      logs: [ [length]: 0 ],
+      receipt_ids: [ 'WKxEdpNqxuSLH5Q9Eu1x99j5KGPqHkPDFwv9s86dgiv', [length]: 1 ],
+      gas_burnt: 2428234030760,
+      tokens_burnt: '242823403076000000000',
+      executor_id: 'multisigtest.testnet',
+      status: {
+        SuccessReceiptId: 'WKxEdpNqxuSLH5Q9Eu1x99j5KGPqHkPDFwv9s86dgiv'
+      }
+    }
+  },
+  receipts_outcome: [
+    {
+      proof: [
+        {
+          hash: '5Dx8KhQ4dMXkBCnWmMuoqxrBZ513JVGJ8pt9UVsrRQTz',
+          direction: 'Left'
+        },
+        [length]: 1
+      ],
+      block_hash: 'GgmsAbDsSjNK25xiV4hnMbLhWtSMzQ36GhnS7C7arN4u',
+      id: 'WKxEdpNqxuSLH5Q9Eu1x99j5KGPqHkPDFwv9s86dgiv',
+      outcome: {
+        logs: [ [length]: 0 ],
+        receipt_ids: [ '83jbmEv7Sref4PXXtwDD7fphDNMxoUE6odQMYWVioene', [length]: 1 ],
+        gas_burnt: 8026429916861,
+        tokens_burnt: '802642991686100000000',
+        executor_id: 'multisigtest.testnet',
+        status: { SuccessValue: 'NA==' }
+      }
+    },
+    {
+      proof: [ [length]: 0 ],
+      block_hash: '51cTk9UUZsXs4NgCACLnSi2jmEWTmyroRmwNTPJWYYf9',
+      id: '83jbmEv7Sref4PXXtwDD7fphDNMxoUE6odQMYWVioene',
+      outcome: {
+        logs: [ [length]: 0 ],
+        receipt_ids: [ [length]: 0 ],
+        gas_burnt: 0,
+        tokens_burnt: '0',
+        executor_id: 'multisigtest.testnet',
+        status: { SuccessValue: '' }
+      }
+    },
+    [length]: 2
+  ]
+}
+```
+There are two receipts generated. `WKxEdpNqxuSLH5Q9Eu1x99j5KGPqHkPDFwv9s86dgiv` is the receipt converted from the transaction
+while the other one is a refund.
+
+- Second step: `confirm`. A second transaction is sent to confirm the transfer. This transaction takes the request id
+returned by the first transaction and does the actual transfer. The transaction result is as follows
+```javascript
+{
+  status: { SuccessValue: '' },
+  transaction: {
+    signer_id: 'multisigtest.testnet',
+    public_key: 'ed25519:BmVX32jhvEd8d8outiQdjf66GGYV3pb7kaxrKTdNisCz',
+    nonce: 5,
+    receiver_id: 'multisigtest.testnet',
+    actions: [
+      {
+        FunctionCall: {
+          method_name: 'confirm',
+          args: 'eyJyZXF1ZXN0X2lkIjo0fQ==',
+          gas: 30000000000000,
```

Comment:
The amount of gas in the last `confirm` call has to be large enough to cover the execution logic of the multi-sig contract PLUS full amount of gas required to issue the intended call. In this case it's a `Transfer` action, which is relatively cheap.

In case of a function call from a multi-sig contract, the amount of gas has to be: `25Tgas + <AMOUNT_OF_ATTACHED_GAS>`   where `<AMOUNT_OF_ATTACHED_GAS>` is the gas attached to outgoing request.
NOTE: That the non-final `confirm` calls (e.g. the first `add_request_and_confirm`) don't require  `<AMOUNT_OF_ATTACHED_GAS>` to be attached, so `25Tgas` for them is enough.